src/python/pants/jvm/bsp/compile.py

```diff
@@ -69,13 +69,12 @@ async def notify_for_classpath_entry(
     return entry
 
 
-@rule_helper
 async def _jvm_bsp_compile(
     request: BSPCompileRequest, classpath_entry_request: ClasspathEntryRequestFactory
 ) -> BSPCompileResult:
     """Generically handles a BSPCompileRequest (subclass).
 
-    This is a `@rule_helper` rather than a `@rule`, because BSP backends like `java` and `scala`
+    This is a rule helper rather than a `@rule`, because BSP backends like `java` and `scala`
```

Comment:
I think that the note on `@union`s in here is no longer relevant with your change! Huzzah.

For another change though obviously.

Reply:
> For another PR though obviously.

thanks for that added clarity :)